### PR TITLE
feat(colors): add hover colors to `@carbon/colors`

### LINKED ISSUE #8822: Add hover color values to the `@carbon/colors` package
We currently make use of the hover colors from the IDL swatches in design specs but are unable to refer to them in code. This issue is to implement these hover colors in code to be used by themes or product teams at IBM.

### PR DESCRIPTION
**Pull request by joshblack**
Closes #8822

This PR updates our `@carbon/colors` package to include the hover palette from the IDL. It also updates our build task to export hover variables in Sass.

#### Changelog

**New**

- Hover colors from the IDL

**Changed**

- Update build taskt to emit hover colors

**Removed**

#### Testing / Reviewing

- Pull down the PR locally
- Make sure to run `yarn build` in `@carbon/colors`
- Verify that the `scss/mixins.scss` file includes the hover colors from the IDL
- Verify that the `index.scss` file includes the hover colors from the IDL

### CODE REVIEW

**Comment by joshblack** [comment]
FYI @sstrubberg here's that draft PR!

**Comment by sstrubberg** [comment]
> FYI @sstrubberg here's that draft PR!

Thx! I got all the colors in. Still getting some build errors though and i'm not confident about the tests either.

**Comment by joshblack** [comment]
cc @aagonzales here are the hover colors added to our colors package from the IDL. We used the values from the file and converted them to HEX

**Review by sstrubberg — APPROVED** [review]
so many colors!

**Review by emyarod — APPROVED** [review]
looks good to me

**Comment by joshblack** [comment]
Also just a heads up @aagonzales I updated the colors preview so the table should include the color hovers in there with the hex value 👍

File: packages/colors/__tests__/__snapshots__/colors-test.js.snap [hunk 1]
```diff
@@ -134,10 +141,167 @@ Object {
...
+`;
+
+exports[`unstable_hoverColors 1`] = `
+Object {
+  "blackHover": "#212121",
+  "blueHover": Object {
+    "10": "#dbebff",
+    "100": "#001f75",
+    "20": "#b8d3ff",
+    "30": "#b8d3ff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
    "30": "#8ab6ff",
```

File: packages/colors/__tests__/__snapshots__/colors-test.js.snap [hunk 2]
```diff
@@ -134,10 +141,167 @@ Object {
...
+    "40": "#9199a1",
+    "50": "#757b85",
+    "60": "#585e64",
+    "70": "#5d646a",
+    "80": "#434a51",
+    "90": "#2b3236",
+  },
+  "cyanHover": Object {
+    "10": "#cceeff",
+    "100": "#cceeff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
    "100": "#0b2947",
```

File: packages/colors/__tests__/__snapshots__/colors-test.js.snap [hunk 3]
```diff
@@ -134,10 +141,167 @@ Object {
...
+    "30": "#ff9d57",
+    "40": "#fa6800",
+    "50": "#cc5500",
+    "60": "#9e4200",
+    "70": "#a84400",
+    "80": "#753300",
+    "90": "#522200",
+  },
+  "purpleHover": Object {
+    "10": "#eee6ff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
    "10": "#ede5ff",
```

File: packages/colors/__tests__/__snapshots__/modules-test.js.snap [hunk 4]
```diff
@@ -147,110 +165,240 @@ Object {
...
   "orange-60": "#ba4e00",
+  "orange-60-hover": "#9e4200",
   "orange-70": "#8a3800",
+  "orange-70-hover": "#a84400",
+  "orange-80": "#5e2900",
+  "orange-80-hover": "#753300",
+  "orange-90": "#3e1a00",
+  "orange-90-hover": "#522200",
   "purple-10": "#f6f2ff",
+  "purple-10-hover": "#eee6ff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/modules-test.js.snap]:
```suggestion
  "purple-10-hover": "#ede5ff",
```

File: packages/colors/__tests__/__snapshots__/modules-test.js.snap [hunk 5]
```diff
@@ -147,110 +165,240 @@ Object {
...
   "cool-gray-70": "#4d5358",
+  "cool-gray-70-hover": "#5d646a",
   "cool-gray-80": "#343a3f",
+  "cool-gray-80-hover": "#434a51",
   "cool-gray-90": "#21272a",
+  "cool-gray-90-hover": "#2b3236",
   "cyan-10": "#e5f6ff",
+  "cyan-10-hover": "#cceeff",
   "cyan-100": "#061727",
+  "cyan-100-hover": "#cceeff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/modules-test.js.snap]:
```suggestion
  "cyan-100-hover": "#0b2947",
```

File: packages/colors/__tests__/__snapshots__/modules-test.js.snap [hunk 6]
```diff
@@ -4,16 +4,27 @@ exports[`@carbon/colors Public API 1`] = `
...
   "black-100": "#000000",
+  "black-hover": "#212121",
   "blue-10": "#edf5ff",
+  "blue-10-hover": "#dbebff",
   "blue-100": "#001141",
+  "blue-100-hover": "#001f75",
   "blue-20": "#d0e2ff",
+  "blue-20-hover": "#b8d3ff",
   "blue-30": "#a6c8ff",
+  "blue-30-hover": "#b8d3ff",
```
Comment (aagonzales) [inline packages/colors/__tests__/__snapshots__/modules-test.js.snap]:
```suggestion
  "blue-30-hover": "#8ab6ff",
```

File: packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap [hunk 7]
```diff
@@ -147,110 +165,240 @@ Object {
...
   "orange-60": "#ba4e00",
+  "orange-60-hover": "#9e4200",
   "orange-70": "#8a3800",
+  "orange-70-hover": "#a84400",
+  "orange-80": "#5e2900",
+  "orange-80-hover": "#753300",
+  "orange-90": "#3e1a00",
+  "orange-90-hover": "#522200",
   "purple-10": "#f6f2ff",
+  "purple-10-hover": "#eee6ff",
```
Comment (aagonzales) [inline packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
  "purple-10-hover": "#ede5ff",
```

File: packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap [hunk 8]
```diff
@@ -4,16 +4,27 @@ exports[`@carbon/styles/scss/colors Public API 1`] = `
...
   "black-100": "#000000",
+  "black-hover": "#212121",
   "blue-10": "#edf5ff",
+  "blue-10-hover": "#dbebff",
   "blue-100": "#001141",
+  "blue-100-hover": "#001f75",
   "blue-20": "#d0e2ff",
+  "blue-20-hover": "#b8d3ff",
   "blue-30": "#a6c8ff",
+  "blue-30-hover": "#b8d3ff",
```
Comment (aagonzales) [inline packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
  "blue-30-hover": "#8ab6ff",
```

File: packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap [hunk 9]
```diff
@@ -147,110 +165,240 @@ Object {
...
   "cool-gray-70": "#4d5358",
+  "cool-gray-70-hover": "#5d646a",
   "cool-gray-80": "#343a3f",
+  "cool-gray-80-hover": "#434a51",
   "cool-gray-90": "#21272a",
+  "cool-gray-90-hover": "#2b3236",
   "cyan-10": "#e5f6ff",
+  "cyan-10-hover": "#cceeff",
   "cyan-100": "#061727",
+  "cyan-100-hover": "#cceeff",
```
Comment (aagonzales) [inline packages/styles/scss/__tests__/__snapshots__/colors-test.js.snap]:
```suggestion
  "cyan-100-hover": "#0b2947",
```

**Review by aagonzales — CHANGES_REQUESTED** [review]
Values that need to be updated: 
Purple-10-hover: `#ede5ff`
Cyan-100-hover: `#0b2947`
Blue-30-hover: `#8ab6ff`

**Comment by joshblack** [comment]
Thanks Anna! Just updated. Just a quick heads up, all the `*.snap` files are generated so feel free to look at the `colors.js` file here: https://github.com/carbon-design-system/carbon/pull/8942/files#diff-e2ecd2a0df3376b7c433e2fdc41d587c14a3b56ce0b6bfa65bb6385246fe69e7 and disregard the rest (should save some time, I hope 🤞 )

**Review by aagonzales — APPROVED** [review]
lol i was dumb and didn't even think to look at the elements preview smh

**Comment by joshblack** [comment]
😂

**Pull request merged**